plugins/module_utils/vm_snapshot.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from __future__ import annotations
+
+__metaclass__ = type
+
+from .rest_client import RestClient
+
+from ..module_utils.utils import PayloadMapper, get_query
+
+from ..module_utils.typed_classes import (
+    TypedVMSnapshotToAnsible,
+    TypedVMSnapshotFromAnsible,
+)
+from typing import List, Any, Dict, Optional
+
+
+class VMSnapshot(PayloadMapper):
+    def __init__(
+        self,
+        uuid: Optional[str] = None,
+        domain_uuid: Optional[str] = None,
```

Comment:
vm_uuid instead

Reply:
`vm_uuid` and `domain_uuid` - are they not the same?

Reply:
Sorry, it came to a minor misunderstanding. I fixed it now :)